File: capitalize-first.js
// Write a recursive function called capitalizeFirst. Given an array of strings, capitalize the first letter of each string in the array.

function capitalizeFirst (array) {
  let newArr = []


  function capitalizeStr(arr) {
    if(arr.length > 0) {
      let newStr = arr[0].charAt(0).toUpperCase() + arr[0].slice(1, arr[0].length)
      newArr.push(newStr)
      capitalizeStr(arr.filter ( str => str != arr[0]))
    }
  }

  capitalizeStr(array)

  return newArr
}

console.log(capitalizeFirst(["hello", "chao", "bye", "daniel"]))

File: capitalize-first.rb
def capitalize_helper (array, new_array)
  if array.length == 0
    return
  end
  new_str =  array[0].capitalize
  new_array.push(new_str)
  array = array.drop(1)
  capitalize_helper(array, new_array)
end


def capitalize_first (array_of_strings)
  new_array = []

  capitalize_helper(array_of_strings, new_array)

  new_array
end

p capitalize_first(["hello", "daniel", "capitalize", "works"])
